### PR TITLE
Update lookForSslCert to be cross-platform

### PR DESCRIPTION
**Pull request by Sean-Der**
Use traverseDirectory from PiC so we are portable. This is part of the
Windows porting.

### CODE REVIEW

**Comment by Sean-Der** [comment]
If I have `/home/sean/foo/test.pem` the follow all work

```
AWS_KVS_CACERT_PATH=/home/sean/foo
AWS_KVS_CACERT_PATH=/home/sean/foo/test.pem
```

File: samples/Common.c [hunk 1]
```diff
@@ -563,16 +563,29 @@ STATUS handleRemoteCandidate(PSampleStreamingSession pSampleStreamingSession, PS
     return retStatus;
 }
 
+STATUS traverseDirectoryPEMFileScan(UINT64 customData, DIR_ENTRY_TYPES entryType, PCHAR fullPath, PCHAR fileName) {
+    PCHAR certName = (PCHAR) customData;
+    INT32 fileNameLen = STRLEN(fileName);
```
Comment (MushMal) [inline samples/Common.c:568]: Strlen returns size_t which will generate warning n some compilers. Use uint32 with a cast perhaps? Or size_t

File: samples/Common.c [hunk 2]
```diff
@@ -563,16 +563,29 @@ STATUS handleRemoteCandidate(PSampleStreamingSession pSampleStreamingSession, PS
...
+
+    if(fileNameLen > ARRAY_SIZE(CA_CERT_PEM_FILE_EXTENSION) + 1 &&
+       (STRCMPI(CA_CERT_PEM_FILE_EXTENSION, &fileName[fileNameLen - ARRAY_SIZE(CA_CERT_PEM_FILE_EXTENSION) + 1]) == 0))
+    {
+        certName[0] = FPATHSEPARATOR;
+        certName++;
+        STRCPY(certName, fileName);
```
Comment (MushMal) [inline samples/Common.c:575]: I think there is a way to terminate the iterative - was it returning error or something

File: samples/Common.c (hunk 1 above)
Comment (Sean-Der) [reply samples/Common.c]: Fixed!

File: samples/Common.c (hunk 2 above)
Comment (Sean-Der) [reply samples/Common.c:575]: I double checked and just a `CHK_STATUS` so I don't believe so? happy to make a backlog item out of it though!

**Pull request merged**